sklearn/covariance/_shrunk_covariance.py

```diff
@@ -106,12 +106,16 @@ def shrunk_covariance(emp_cov, shrinkage=0.1):
 
     where `mu = trace(cov) / n_features`.
     """
-    emp_cov = check_array(emp_cov)
-    n_features = emp_cov.shape[0]
+    emp_cov = check_array(emp_cov, allow_nd=True)
+    n_features = emp_cov.shape[-1]
+    if not 0 <= shrinkage <= 1:
+        raise ValueError("shrinkage must be in [0, 1] (Got %d)" % shrinkage)
```

Comment:
Actually, we already have the parameter validation with the decorator `@valid_params` already.

So we should not need this.

Reply:
Ok, I removed it.